scripts/mvf_rechunk.py

```diff
@@ -0,0 +1,231 @@
+    parser.add_argument('--new-prefix', help='Replacement for capture block ID in output bucket names')
+    parser.add_argument('source', help='Input .rdb file')
+    parser.add_argument('dest', help='Output directory')
+    parser.add_argument('spec', nargs='*', default=[], type=RechunkSpec,
+                        metavar='STREAM/ARRAY:TIME,FREQ', help='New chunk specification')
+    args = parser.parse_args()
+    return args
+
+
+def get_streams(telstate, streams):
+    """Determine streams to copy based on what the user asked for"""
+    archived_streams = telstate.get('sdp_archived_streams', [])
+    if not archived_streams:
+        raise RuntimeError('Source dataset does not contain any streams')
+    if streams is None:
+        streams = archived_streams
+    else:
+        for stream in streams:
+            if stream not in archived_streams:
+                raise RuntimeError('Stream {!r} is not known (should be one of {})'
+                                   .format(stream, ', '.join(archived_streams)))
+
+    return streams
+
+
+def main():
+    args = parse_args()
+    dask.config.set(num_workers=args.workers)
+
+    # Lightweight open with no data - just to create telstate and identify the CBID
+    ds = TelstateDataSource.from_url(args.source, upgrade_flags=False, chunk_store=None)
+    # View the CBID, but not any specific stream
+    cbid = ds.capture_block_id
+    telstate = ds.telstate.root().view(cbid)
+    streams = get_streams(telstate, args.streams)
+
+    # Find all arrays in the selected streams, and also ensure we're not
+    # trying to write things back on top of an existing dataset.
+    arrays = {}
+    for stream_name in streams:
+        sts = view_capture_stream(telstate, cbid, stream_name)
+        try:
+            chunk_info = sts['chunk_info']
+        except KeyError as exc:
+            raise RuntimeError('Could not get chunk info for {!r}: {}'.format(stream_name, exc))
+        for array_name, array_info in chunk_info.items():
+            if args.new_prefix is not None:
+                array_info['prefix'] = args.new_prefix + '-' + stream_name.replace('_', '-')
+            prefix = array_info['prefix']
+            path = os.path.join(args.dest, prefix)
+            if os.path.exists(path):
+                raise RuntimeError('Directory {!r} already exists'.format(path))
+            store = get_chunk_store(args.source, sts, array_name)
+            # Older files have dtype as an object that can't be encoded in msgpack
+            dtype = np.dtype(array_info['dtype'])
+            array_info['dtype'] = np.lib.format.dtype_to_descr(dtype)
+            arrays[(stream_name, array_name)] = Array(stream_name, array_name, store, array_info)
+
+    # Apply DATA_LOST bits to the flags arrays. This is a less efficient approach than
+    # datasources.py, but much simpler.
+    for stream_name in streams:
+        flags_array = arrays.get((stream_name, 'flags'))
+        if flags_array:
```

Comment:
I prefer the following to reduce indentation (I see it's already used to good effect further into the loops):

```Python
if not flags_array:
    continue
```

Reply:
Done.